src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -840,54 +789,44 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
-    if (image.size() == 1)
-      auto_flash(workingDevice, flashType, image.front());
-    
-    if (image.size() == 2) {
-      std::cout << "CAUTION! Force flashing the platform on the device without any checks." <<
-                   "Please make sure that the correct information is passed in." << std::endl;
-      auto image_paths = find_flash_image_paths(image);
-      update_shell(workingDevice->get_device_id(), flashType, image_paths.front(), (image_paths.size() == 2 ? image_paths[1]: ""));
-    }
-    
-    if (image.size() > 2)
-      throw xrt_core::error("Please specify either 1 or 2 flash images");
-    return;
-  }
-
   if (!update.empty()) {
-    XBU::verbose("Sub command: --base");
-    XBUtilities::sudo_or_throw("Root privileges are required to update the devices flash image");
-    if (update.compare("all") == 0)
-      // Note: To get around a bug in the SC flashing code base,
-      //       auto_flash will clear the collection. This code need to be refactored and clean up.
-      auto_flash( workingDevice );
-    else {
-      if (update.compare("flash") == 0)
-        throw xrt_core::error("Platform only update is not supported");
-
-      if (update.compare("sc") == 0)
-        throw xrt_core::error("SC only update is not supported");
-       
-      throw xrt_core::error("Please specify a valid value");
+    // Get a list of images known exist
+    const auto validated_images = find_flash_image_paths(image);
+    if (validated_images.size() <= 2) {
+      // Get the device
+      auto & workingDevice = deviceCollection[0];
+      XBU::verbose("Sub command: --base");
+      XBUtilities::sudo_or_throw("Root privileges are required to update the devices flash image");
+      if (update.compare("all") == 0) {
+        if (validated_images.empty())
+          auto_flash(workingDevice);
+        else if(boost::filesystem::extension(validated_images[0]).compare(".xsabin") == 0)
+          auto_flash(workingDevice, flashType, validated_images[0]);
+        else
+          throw xrt_core::error("Please provide a valid xsabin file or specify a specific base to flash");
+      }
+      else if (update.compare("sc") == 0) {
+          if (!validated_images.empty())
+            update_SC(workingDevice.get()->get_device_id(), validated_images[0]);
+          else
+            throw xrt_core::error("Usage: xbmgmt program --device='0000:00:00.0' --base sc --image='/path/to/flash_image");
+      }
+      else if (update.compare("shell") == 0) {
+          if (!validated_images.empty())
+            update_shell(workingDevice.get()->get_device_id(), validated_images, flashType);
+          else
+            throw xrt_core::error("Usage: xbmgmt program --device='0000:00:00.0' --base shell --image='/path/to/flash_image OR shell_name'");
+      }
+      else
+        throw xrt_core::error("Please specify a valid value");
     }
+    else
+      throw xrt_core::error("Please specify either 1 or 2 flash images");
```

Comment:
I can format out the middle 2 if statements into a separate function, but, I do not think that would help with readability. I am open to any changes though

Reply:
@AShivangi That is a very good point. For example the LLVM coding style favors early exit to avoid this.
For example here you could have just very early:
```c++
if (validated_images.size() > 2) { 
       throw xrt_core::error("Please specify only 1 or 2 flash images"); 
```
and then you do not need the `else`. :-)
You can generalize the idea everywhere with `throw`/`break`/`return`...

Reply:
I was able to get remove all but the `all` condition statement. And removed the images validation statement to cause an earlier exit.